Line area type codec causing `EncoderException`

### How to reproduce the bug

Use the GPS Area Tool in Line mode. The problem occurs when closing the GUI.

### Additional details

```
[Netty Client IO #6/ERROR] [minecraft/Connection]: Exception caught in connection
io.netty.handler.codec.EncoderException: Failed to encode packet 'serverbound/minecraft:custom_payload'
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:55) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:14) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/neoforge@21.1.93/net.neoforged.neoforge.network.filters.GenericPacketSplitter.encode(GenericPacketSplitter.java:104) ~[neoforge-21.1.93-universal.jar%23595!/:?] {re:classloading}
	at TRANSFORMER/neoforge@21.1.93/net.neoforged.neoforge.network.filters.GenericPacketSplitter.encode(GenericPacketSplitter.java:39) ~[neoforge-21.1.93-universal.jar%23595!/:?] {re:classloading}
	at MC-BOOTSTRAP/io.netty.codec@4.1.97.Final/io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:90) ~[netty-codec-4.1.97.Final.jar%23130!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.ChannelOutboundHandlerAdapter.write(ChannelOutboundHandlerAdapter.java:113) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {re:computing_frames,re:mixin}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.Connection$2.write(Connection.java:528) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:940) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:966) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:934) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:984) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1025) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:306) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.Connection.doSendPacket(Connection.java:364) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:mixin,re:classloading,pl:mixin:APP:connectivity.mixins.json:AdvancedPacketErrorLogging from mod connectivity,pl:mixin:APP:mixins/neoforge/nochatreports-neoforge.mixins.json:client.MixinClientConnection from mod nochatreports,pl:mixin:APP:connectivity.mixins.json:ConnectionMixin from mod connectivity,pl:mixin:APP:jamlib.mixins.json:event.ConnectionMixin from mod jamlib,pl:mixin:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.Connection.lambda$sendPacket$12(Connection.java:359) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:mixin,re:classloading,pl:mixin:APP:connectivity.mixins.json:AdvancedPacketErrorLogging from mod connectivity,pl:mixin:APP:mixins/neoforge/nochatreports-neoforge.mixins.json:client.MixinClientConnection from mod nochatreports,pl:mixin:APP:connectivity.mixins.json:ConnectionMixin from mod connectivity,pl:mixin:APP:jamlib.mixins.json:event.ConnectionMixin from mod jamlib,pl:mixin:A}
	at MC-BOOTSTRAP/io.netty.common@4.1.97.Final/io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.97.Final.jar%23131!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.common@4.1.97.Final/io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.97.Final.jar%23131!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.common@4.1.97.Final/io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.97.Final.jar%23131!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.transport@4.1.97.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569) ~[netty-transport-4.1.97.Final.jar%23136!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.common@4.1.97.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.97.Final.jar%23131!/:4.1.97.Final] {}
	at MC-BOOTSTRAP/io.netty.common@4.1.97.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.97.Final.jar%23131!/:4.1.97.Final] {}
	at java.base/java.lang.Thread.run(Unknown Source) [?:?] {re:mixin}
Caused by: java.lang.RuntimeException: Failed encoding custom payload pneumaticcraft:update_gps_area_tool: java.lang.IllegalStateException: Can't encode 'line', expected 'null'
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.common.custom.CustomPacketPayload$1.writeCap(CustomPacketPayload.java:42) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.common.custom.CustomPacketPayload$1.encode(CustomPacketPayload.java:47) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.common.custom.CustomPacketPayload$1.encode(CustomPacketPayload.java:29) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$4.encode(StreamCodec.java:81) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$5.encode(StreamCodec.java:97) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$5.encode(StreamCodec.java:87) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:53) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	... 26 more
Caused by: java.lang.IllegalStateException: Can't encode 'line', expected 'null'
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$3.encode(StreamCodec.java:57) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$6.encode(StreamCodec.java:117) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$12.encode(StreamCodec.java:269) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$8.encode(StreamCodec.java:156) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.common.custom.CustomPacketPayload$1.writeCap(CustomPacketPayload.java:40) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.common.custom.CustomPacketPayload$1.encode(CustomPacketPayload.java:47) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.common.custom.CustomPacketPayload$1.encode(CustomPacketPayload.java:29) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$4.encode(StreamCodec.java:81) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$5.encode(StreamCodec.java:97) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.StreamCodec$5.encode(StreamCodec.java:87) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:53) ~[client-1.21.1-20240808.144430-srg.jar%23594!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	... 26 more
```

### Which Minecraft version are you using?

1.21

### Which version of PneumaticCraft: Repressurized are you using?

8.2.6

--- a/src/main/java/me/desht/pneumaticcraft/common/drone/progwidgets/area/AreaTypeLine.java
+++ b/src/main/java/me/desht/pneumaticcraft/common/drone/progwidgets/area/AreaTypeLine.java
@@ -30,12 +30,12 @@ import net.minecraft.world.phys.Vec3;
 import java.util.function.Consumer;
 
 public class AreaTypeLine extends AreaType {
+    private static final AreaTypeLine INSTANCE = new AreaTypeLine();
+
     public static final MapCodec<AreaTypeLine> CODEC = MapCodec.unit(AreaTypeLine.INSTANCE);
     public static final StreamCodec<FriendlyByteBuf, AreaTypeLine> STREAM_CODEC = StreamCodec.unit(AreaTypeLine.INSTANCE);
 
     public static final String ID = "line";
-
-    private static final AreaTypeLine INSTANCE = new AreaTypeLine();
 
     public static AreaTypeLine instance() {
         return INSTANCE;
@@ -47,7 +47,7 @@ public class AreaTypeLine extends AreaType {
 
     @Override
     public AreaType copy() {
-        return new AreaTypeLine();
+        return INSTANCE;
     }
 
     @Override

